adapters/mobfoxpb/mobfoxpb.go

```diff
@@ -0,0 +1,130 @@
+package mobfoxpb
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/buger/jsonparser"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// MobfoxpbAdapter adapter structure
+type MobfoxpbAdapter struct {
+	URI string
+}
+
+// NewMobfoxpbBidder Initializes the Bidder
+func NewMobfoxpbBidder(endpoint string) *MobfoxpbAdapter {
+	return &MobfoxpbAdapter{
+		URI: endpoint,
+	}
+}
+
+// MakeRequests create bid request for mobfoxpb demand
+func (a *MobfoxpbAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var err error
+	var tagID string
+
+	var adapterRequests []*adapters.RequestData
+
+	reqCopy := *request
+	for _, imp := range request.Imp {
```

Comment:
Some thoughts on the for loop inside `MakeRequests`: I read above that you expect arrays with a single `Imp`. Even if that's always the case, I left some comments below:
```
28   func (a *MobfoxpbAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
29   *--  5 lines: var errs []error--------------------------------------------------------------------------
34  
35       reqCopy := *request
36       for _, imp := range request.Imp {
   +         // No need to manually create a local copy, the foreach statement creates a local copy of the current element
37 -         reqCopy.Imp = []openrtb.Imp{imp}
38  
39 -         tagID, err = jsonparser.GetString(reqCopy.Imp[0].Ext, "bidder", "TagID")
   +         // Unmarshalling the current element is safe
   +         tagID, err = jsonparser.GetString(imp.Ext, "bidder", "TagID")
40           if err != nil {
41               errs = append(errs, err)
42               continue
43           }
44  
45 -         reqCopy.Imp[0].TagID = tagID
   +         // In golang, the foreach statement creates a local copy of the elements of the array
   +         // it is traversing. Therefore, it's safe to overwrite `TagID`
   +         imp.TagID = tagID
   +         // It might be better to move the `Imp` slice creation down here in order to do it only after after we
   +         // made sure `jsonparser.GetString(reqCopy.Imp[0].Ext, "bidder", "TagID")` did not return an error
   +         reqCopy.Imp = []openrtb.Imp{imp}
46  
47           adapterReq, err := a.makeRequest(&reqCopy)
48           if err != nil {
49               errs = append(errs, err)
50           }
51           if adapterReq != nil {
52               adapterRequests = append(adapterRequests, adapterReq)
   +             // Given that we succeeded, we should break here, isn't it? Unless, we want 
   +             // to append more than one "valid" `adapterReq` in the `adapterRequests` array
   +             break
53           }
54       }
55       return adapterRequests, errs
56   }
adapters/mobfoxpb/mobfoxpb.go
```

Reply:
Done